dbt/adapters/duckdb/connections.py

```diff
@@ -38,13 +38,42 @@ class DuckDBCredentials(Credentials):
     # in this dbt project; defaults to "." (the current working directory)
     external_root: str = "."
 
+    # identify whether to use the default credential provider chain for AWS/GCloud
+    # instead of statically defined environment variables
+    use_credential_provider: Optional[str] = None
+
     @property
     def type(self):
         return "duckdb"
 
     def _connection_keys(self):
         return ("database", "schema", "path")
 
+    def load_settings(self) -> Dict[str, str]:
+        settings = self.settings or {}
+        if self.use_credential_provider:
+            if self.use_credential_provider == "aws":
+                settings.update(_load_aws_credentials())
+            else:
+                raise ValueError(
+                    "Unsupported value for use_credential_provider: "
+                    + self.use_credential_provider
+                )
+        return settings
+
+
+def _load_aws_credentials() -> Dict[str, Any]:
+    import boto3.session
+
+    session = boto3.session.Session()
+    aws_creds = session.get_credentials().get_frozen_credentials()
```

Comment:
I would check whether the session is valid by calling get_caller_identity(). This way users get a clear error message when their session is expired/invalid credentials instead of just unauthorized when they try to access data while running their models.